src/pipdeptree/_models/package.py

```diff
@@ -23,13 +23,14 @@ class Package(ABC):
 
     UNKNOWN_LICENSE_STR = "(Unknown license)"
 
-    def __init__(self, obj: DistInfoDistribution) -> None:
-        self._obj: DistInfoDistribution = obj
-        self.key = pep503_normalize(obj.key)
+    def __init__(self, obj: Distribution) -> None:
+        self._obj: Distribution = obj
+        self.key = pep503_normalize(obj.name)
+        self.editable = False
 
     @property
     def project_name(self) -> str:
-        return self._obj.project_name  # type: ignore[no-any-return]
+        return self._obj.name  # type: ignore[no-any-return]
```

Comment:
Because of what was discussed [here](https://github.com/tox-dev/pipdeptree/pull/333#discussion_r1528986917), this could be rewritten like so:
```suggestion
        return self._obj.metadata['Name']
```